graphql_compiler/compiler/compiler_frontend.py

```diff
@@ -812,10 +813,92 @@ def _preprocess_graphql_string(graphql_string):
+    schema -- all directives supported by the compiler must be declared by the schema, regardless of
+    whether each directive is used in the query or not.
+
+    Args:
+        schema: GraphQL schema object, created using the GraphQL library
+        ast: abstract syntax tree representation of a graphql query
+
+    Returns:
+        list containing schema and/or query validation errors
+    """
+    core_graphql_errors = validate(schema, ast)
+
+    # The following directives appear in the core-graphql library, but are not supported by the
+    # graphql compiler.
+    unsupported_default_directives = frozenset([
+        frozenset([
+            'include',
+            frozenset(['FIELD', 'FRAGMENT_SPREAD', 'INLINE_FRAGMENT']),
+            frozenset(['if'])
+        ]),
+        frozenset([
+            'skip',
+            frozenset(['FIELD', 'FRAGMENT_SPREAD', 'INLINE_FRAGMENT']),
+            frozenset(['if'])
+        ]),
+        frozenset([
+            'deprecated',
+            frozenset(['ENUM_VALUE', 'FIELD_DEFINITION']),
+            frozenset(['reason'])
+        ])
+    ])
+
+    # Extract name, locations and args keys in order to compare schema directives and directives
+    # which are supported by the graphql compiler.
+
+    # Directives expected by the graphql compiler.
+    expected_directives = {
+        frozenset([
+            directive.name,
+            frozenset(directive.locations),
+            frozenset(six.viewkeys(directive.args))
+        ])
+        for directive in DIRECTIVES
+    }
+
+    # Graphql-core automatically injects default directives into the schema, regardless of whether
+    # the schema supports said directives. Hence, while the directives contained in
+    # unsupported_default_directives are incompatible with the graphql-compiler, we still expect
+    # them in the parsed schema string.
+    expected_directives.update(unsupported_default_directives)
+
+    # Directives provided in the parsed graphql schema.
+    actual_directives = {
+        frozenset([
+            directive.name,
+            frozenset(directive.locations),
+            frozenset(six.viewkeys(directive.args))
+        ])
+        for directive in schema.get_directives()
+    }
+
+    # Directives missing from the actual directives provided.
+    missing_directives = expected_directives - actual_directives
+    if missing_directives:
+        missing_message = (u'The following directives were missing from the '
+                           u'provided schema: {}'.format(missing_directives))
+        core_graphql_errors.append(missing_message)
+
+    # Directives that are not specified by the core graphql library.
+    extra_directives = actual_directives - expected_directives
+    if extra_directives:
+        extra_message = (u'The following directives were supplied in the given schema, but are not '
+                         u'defined in the core graphql library: {}'.format(extra_directives))
```

Comment:
I think this error should say `not supported by the compiler` rather than mentioning the core GraphQL library. Also, always capitalize `graphql` -> `GraphQL`.